Sun Mar 10 00:18:03 FET 2013  Sergei Trofimovich <slyfox@community.haskell.org>
  * Setup.hs: switch to base-4 exceptions
  
  Fixes build failure on ghc-7.6:
      $ runhaskell Setup.hs configure
      Setup.hs:394:14: Not in scope: `E.try'
  
  base-4 is available/default since ghc-6.10.
diff -rN -u old-GF/Setup.hs new-GF/Setup.hs
--- old-GF/Setup.hs	2013-03-10 00:42:24.440641383 +0300
+++ new-GF/Setup.hs	2013-03-10 00:42:24.442641383 +0300
@@ -9,7 +9,7 @@
 import Data.List(isPrefixOf,intersect)
 import Data.Maybe(listToMaybe)
 import System.IO
-import qualified System.IO.Error as E
+import qualified Control.Exception as E
 import System.Cmd
 import System.FilePath
 import System.Directory
@@ -18,6 +18,9 @@
 
 import WebSetup
 
+tryIOE :: IO a -> IO (Either E.IOException a)
+tryIOE = E.try
+
 main :: IO ()
 main = defaultMainWithHooks simpleUserHooks{ preBuild =gfPreBuild
                                            , postBuild=buildRGL
@@ -391,7 +394,7 @@
 
 -- | Create autogen module with detailed version info by querying darcs
 extractDarcsVersion distFlag =
-  do info <- E.try askDarcs
+  do info <- tryIOE askDarcs
      createDirectoryIfMissing True autogenPath
      updateFile versionModulePath $ unlines $
        ["module "++modname++" where",
@@ -410,7 +413,7 @@
                       tag:_ -> ["--from-tag="++tag]
          changes <- lines `fmap` readProcess "darcs" ("changes":from) ""
          let dates = init' (filter ((`notElem` [""," "]).take 1) changes)
-         whatsnew<-E.try $ lines `fmap` readProcess "darcs" ["whatsnew","-s"] ""
+         whatsnew <- tryIOE $ lines `fmap` readProcess "darcs" ["whatsnew","-s"] ""
          return (listToMaybe tags,listToMaybe dates,
                  length dates,either (const 0) length whatsnew)
 
@@ -419,7 +422,7 @@
 
 -- | Only update the file if contents has changed
 updateFile path new =
-  do old <- E.try $ readFile path
+  do old <- tryIOE $ readFile path
      when (Right new/=old) $ seq (either (const 0) length old) $
                                  writeFile path new
 
